app/controllers/mixins/ems_common/angular.rb

```diff
@@ -137,7 +137,16 @@ def get_task_args(ems)
           end
         when 'ManageIQ::Providers::Amazon::CloudManager'
           uri = URI.parse(WEBrick::HTTPUtils.escape(params[:default_url]))
-          [user, password, :EC2, params[:provider_region], ems.http_proxy_uri, true, uri]
+          [
+            user,
+            password,
+            :EC2,
+            params[:provider_region],
+            ems.http_proxy_uri,
+            true,
+            uri,
+            :assume_role => params[:service_account].presence,
```

Comment:
hmm.. the UI would send this as `params[:default_service_account]` since it's coming from the default tab.
(Note: When there are no tabs, the `default` prefix is added to the fields)

Reply:
Did you try to run the code? It works as it should, without "default".

Reply:
I did try your code in it's current state and noticed that Add button was disabled -- so could not get past the frontend, to verify this backend change.

In general, the section in the UI under "Endpoints" is associated with a prefix, which matches the tab.
The prefix in this case is `default`, since the tab is presumed to be default

Reply:
> I did try your code in it's current state and noticed that Add button was disabled -- so could not get past the frontend, to verify this backend change.

"Add" button disabled unless you'll do the validation. There is a "Validate" button for that.

Reply:
Add button stays disabled despite a successful validation.

It stays disabled because the `ManageIQ.angular.scope.angularForm.$error.validationRequired` object is populated.
After a successful validation, the `$error` object needs to be empty.

The reason for that is in the JS code there is no variable that holds the ARN value. 
This seems like an oversight, because we do need a variable there.
Also the post validation state is not getting registered, hence the `validationRequired` error.